drivers/regulator/regulator_shell.c

```diff
@@ -364,6 +364,29 @@ static int cmd_errors(const struct shell *sh, size_t argc, char **argv)
 	return 0;
 }
 
+static int cmd_dvsset(const struct shell *sh, size_t argc, char **argv)
+{
+	const struct device *dev;
+	int ret;
+	regulator_dvs_state_t state;
+
+	dev = device_get_binding(argv[1]);
+	if (dev == NULL) {
+		shell_error(sh, "Regulator device %s not available", argv[1]);
+		return -ENODEV;
+	}
+
+	state = (regulator_dvs_state_t)strtol(argv[2], NULL, 10);
```

Comment:
You could use shell_strtoul here to avoid the sign cast and capture any errors.

Reply:
Thanks, was not aware of this.